src/engraving/libmscore/paste.cpp

```diff
@@ -1031,6 +1031,60 @@ void Score::pasteSymbols(XmlReader& e, ChordRest* dst)
     }                                     // inner while readNextstartElement()
 }                                         // pasteSymbolList()
 
+static ChordRest* replaceWithRest(ChordRest* target)
+{
+    target->score()->undoRemoveElement(target);
+    return target->score()->addRest(target->segment(), target->track(), target->ticks(), target->tuplet());
+}
+
+static Note* prepareTarget(ChordRest* target, Note* with, const Fraction& duration)
+{
+    if (!target->segment()->element(target->track())) {
+        return nullptr; // target was removed by previous operation, ignore this
+    }
+    if (target->isChord() && target->ticks() > duration) {
+        target = replaceWithRest(target); // prevent unexpected note splitting
+    }
+    Segment* segment = target->segment();
+    if (segment->measure()->isMMRest()) {
+        Measure* m = segment->measure()->mmRestFirst();
+        segment = m->findSegment(SegmentType::ChordRest, m->tick());
+    }
+    segment = target->score()->setNoteRest(segment, target->track(),
+                                           with->noteVal(), duration, Direction::AUTO, false, {}, false, &target->score()->inputState());
+    return toChord(segment->nextChordRest(target->track()))->upNote();
+}
+
+static Element* prepareTarget(Element* target, Note* with, const Fraction& duration)
+{
+    if (target->isNote() && toNote(target)->chord()->ticks() != duration) {
+        return prepareTarget(toNote(target)->chord(), with, duration);
+    }
+    if (target->isChordRest() && toChordRest(target)->ticks() != duration) {
+        return prepareTarget(toChordRest(target), with, duration);
+    }
+    return target;
+}
+
+static bool canPasteStaff(XmlReader& reader, const Fraction& scale)
```

Comment:
to prevent crashing when using "paste half length" on a range with 1/1024th notes, a bit kludgy but I tried a few other ways and this was the more reliable and least code by far